<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CrossEditorUI</class>
 <widget class="QMainWindow" name="CrossEditorUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>956</width>
    <height>647</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Cross Editor</string>
  </property>
  <property name="dockNestingEnabled">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="central_layout">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="GLHandler" name="gl_handler">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menu">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>956</width>
     <height>38</height>
    </rect>
   </property>
   <widget class="QMenu" name="file_menu">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="new_scene_action"/>
    <addaction name="save_scene_as_action"/>
    <addaction name="setup_project_directory_action"/>
   </widget>
   <widget class="QMenu" name="view_menu">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="file_explorer_action"/>
    <addaction name="scene_explorer_action"/>
    <addaction name="properties_action"/>
   </widget>
   <addaction name="file_menu"/>
   <addaction name="view_menu"/>
  </widget>
  <widget class="QDockWidget" name="file_explorer_dock">
   <property name="windowTitle">
    <string>File Explorer</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="file_explorer_layout">
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>1</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="FileExplorer" name="file_explorer">
       <property name="editTriggers">
        <set>QAbstractItemView::SelectedClicked</set>
       </property>
       <property name="dragEnabled">
        <bool>true</bool>
       </property>
       <property name="dragDropMode">
        <enum>QAbstractItemView::DragOnly</enum>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::ExtendedSelection</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="ProjectDirectoryLabel" name="project_directory_laybel">
       <property name="font">
        <font>
         <pointsize>7</pointsize>
        </font>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="text">
        <string>ProjectDirectory</string>
       </property>
      </widget>
     </item>
    </layout>
    <zorder>project_directory_laybel</zorder>
    <zorder>file_explorer</zorder>
   </widget>
  </widget>
  <widget class="QDockWidget" name="scene_explorer_dock">
   <property name="windowTitle">
    <string>Scene Explorer</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="scene_explorer_layout">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="SceneExplorer" name="scene_explorer">
       <property name="editTriggers">
        <set>QAbstractItemView::SelectedClicked</set>
       </property>
       <property name="dragDropMode">
        <enum>QAbstractItemView::DropOnly</enum>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="uniformRowHeights">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="PropertiesView" name="properties_view">
   <property name="windowTitle">
    <string>Properties</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="properties_view_layout">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>4</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <spacer name="properties_view_spacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionLoad_Scene">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Load Scene</string>
   </property>
  </action>
  <action name="file_explorer_action">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>File Explorer</string>
   </property>
  </action>
  <action name="scene_explorer_action">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Scene Explorer</string>
   </property>
  </action>
  <action name="save_scene_as_action">
   <property name="text">
    <string>Save Scene As</string>
   </property>
  </action>
  <action name="new_scene_action">
   <property name="text">
    <string>New Scene</string>
   </property>
  </action>
  <action name="setup_project_directory_action">
   <property name="text">
    <string>Setup Project Directory</string>
   </property>
  </action>
  <action name="actionNewShader">
   <property name="text">
    <string>New Shader</string>
   </property>
  </action>
  <action name="properties_action">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Properties</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLHandler</class>
   <extends>QOpenGLWidget</extends>
   <header>GLHandler.h</header>
  </customwidget>
  <customwidget>
   <class>FileExplorer</class>
   <extends>QTreeView</extends>
   <header>FileExplorer.h</header>
  </customwidget>
  <customwidget>
   <class>SceneExplorer</class>
   <extends>QTreeView</extends>
   <header>SceneExplorer.h</header>
  </customwidget>
  <customwidget>
   <class>PropertiesView</class>
   <extends>QDockWidget</extends>
   <header>PropertiesView.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ProjectDirectoryLabel</class>
   <extends>QLabel</extends>
   <header>FileExplorer.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../CrossEditor.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>file_explorer_action</sender>
   <signal>toggled(bool)</signal>
   <receiver>file_explorer_dock</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>127</x>
     <y>341</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>properties_action</sender>
   <signal>toggled(bool)</signal>
   <receiver>properties_view</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1153</x>
     <y>648</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scene_explorer_action</sender>
   <signal>toggled(bool)</signal>
   <receiver>scene_explorer_dock</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>127</x>
     <y>955</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>file_explorer_dock</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>file_explorer_action</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>127</x>
     <y>341</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>properties_view</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>properties_action</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1153</x>
     <y>648</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scene_explorer_dock</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>scene_explorer_action</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>127</x>
     <y>955</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
